### PR TITLE
[PAPI] Retry steams upon not receiving data for 10 seconds

### PR DESCRIPTION
**Pull request by filiptronicek**
## Description

This change tries to improve our debugging situation with streams, because [streaming is hard](https://connectrpc.com/docs/go/streaming/#:~:text=comes%20with%20many-,drawbacks%3A,-It%20requires%20excellent).

## Related Issue(s)

Fixes ENT-753

## How to test

https://ft-debug-ub7eaf498b5.preview.gitpod-dev.com/workspaces

1. Make sure that prebuild status watching works (start a new prebuild)
2. Make sure that workspace status streaming does not time out (wait 10s on the workspace list page and see console)

### CODE REVIEW

**Comment by filiptronicek** [comment]
I'm getting this for some reason on a prebuild detail page:
> public-api.ts:282 Uncaught Error: Attempt Timeout: No initial data received within 10 seconds
    at public-api.ts:282:31

This is from `WorkspaceService/WatchWorkspaceStatus`. I wonder what that's doing on the prebuild page 🤔 

cc @geropl

File: components/dashboard/src/service/public-api.ts [hunk 1]
```diff
@@ -275,12 +275,24 @@ export function stream<Response>(
     (async () => {
         while (!abort.signal.aborted) {
             try {
+                let dataReceived = false;
```
Comment (geropl) [inline components/dashboard/src/service/public-api.ts:278]: @filiptronicek I created https://github.com/gitpod-io/gitpod/pull/20176/files to use a class `Timeout` I needed for a similar problem in the frontend.

**Review by geropl — APPROVED** [review]
Code LGTM, (you) tested and it works :wink: :heavy_check_mark:

**Comment by geropl** [comment]
@filiptronicek Anything needs to happen here, beyond the un-hold?
Happy to test again if that helps! :pray:

**Comment by filiptronicek** [comment]
@geropl was just about to merge this. Re-tested a couple of minutes back and added [`f2ab124` (#20172)](https://github.com/gitpod-io/gitpod/pull/20172/commits/f2ab124afb5122b68813628ac183d698c446548c) to make this work with the feature flag in either state.

**Pull request merged**